core/src/test/resources/sql/measure.iq

```diff
@@ -179,6 +179,53 @@ from empm;
 
 !ok
 
+# [CALCITE-6559] LogicalCorrelate Type mismatch
+# Occurs when AVG measure is applied to SMALLINT or TINYINT column.
+# A CAST is added internally to the result of SUM / COUNT, but this cast
```

Comment:
It's difficult to tell from this comment whether this was the cause of the bug or whether this is the new expectation after fixing.  Have you considered moving this comment to CALCITE-6559 instead?